echoregions/regions2d/regions2d.py

```diff
@@ -73,42 +73,140 @@ def to_json(self, save_path: str = None) -> None:
+            Datetime range for expected output of subselected DataFrame. 1st
+            index value must be later than 0th index value.
+        depth_range: List of 2 floats.
+            Depth range for expected output of subselected DataFrame. 1st
+            index value must be larger than 0th index value.
         copy : bool
             Return a copy of the `data` DataFrame
         Returns
         -------
         DataFrame
             A DataFrame subselected from Regions2D.data.
-            There is a row for each region id provided by the region parameter.
+            There is a row for each region id provided by the region_id parameter,
+            and each row has time and depth within or on the boundaries passed
+            in by the time_range and depth_range values.
         """
-        if region is not None:
-            if isinstance(region, DataFrame):
-                region = list(region.region_id)
-            elif isinstance(region, Series):
-                region = [region.region_id]
-            elif (
-                isinstance(region, float)
-                or isinstance(region, int)
-                or isinstance(region, str)
+        region = None
+        untouched = True
+        if region_id is not None:
+            if (
+                isinstance(region_id, float)
+                or isinstance(region_id, int)
+                or isinstance(region_id, str)
             ):
-                region = [region]
-            elif not isinstance(region, list):
+                region_id = [region_id]
+            elif not isinstance(region_id, list):
                 raise TypeError(
-                    f"Invalid Region Type: {type(region)}. Must be \
-                                of type float, str, list, Series, DataFrame, ``None``"
+                    f"Invalid region_id type: {type(region_id)}. Must be \
+                                of type float, int, str, list, ``None``."
                 )
             # Select row by column id
-            region = self.data[self.data["region_id"].isin(region)]
-        else:
+            for value in region_id:
+                if not isinstance(value, (float, int, str)):
+                    raise TypeError(
+                        f"Invalid element in list region_id. Is of \
+                            type: {type(value)}Must be \
+                            of type float, int, str."
+                    )
+            region = self.data[self.data["region_id"].isin(region_id)]
+            untouched = False
+        if time_range is not None:
+            if isinstance(time_range, List):
+                if len(time_range) == 2:
+                    if isinstance(time_range[0], Timestamp) and isinstance(
+                        time_range[1], Timestamp
+                    ):
+                        if time_range[0] < time_range[1]:
+                            if region is None:
+                                region = self.data
```

Comment:
This assignment is hidden and individually assigned in each of the blocks when you test for `region_id`, `time_range` and `depth_range`. I recommend that you move this out and do the assignment upfront (probably with `.copy()` so that you do not modify `self.data` by default) and do modification in each block so that it is easier to keep track of what's what. This way you do not need `untouched` either -- if it is untouched, just assign it back to `self.data`. The dataframe is small in general so I wouldn't be too worried about efficiency just yet.

Reply:
I still think they should have the option to modify the actual inner dataframe, so I'll still have that if copy is false, then we have that the selection modifies the inner dataframe, and as such the class. But I'll set the copy as True so the user does not automatically do this, and instead just receives a copy of the inner dataframe.

Reply:
Yeah I wasn't suggesting removing that option, so your changes look good to me!